rust/hyperlane-core/src/types/checkpoint.rs

```diff
@@ -33,18 +51,47 @@ impl Debug for Checkpoint {
     }
 }
 
+impl Debug for CheckpointWithMessageId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "CheckpointWithMessageId {{ checkpoint: {:?}, message_id: {:?} }}",
+            self.checkpoint, self.message_id
+        )
+    }
+}
+
 #[async_trait]
 impl Signable for Checkpoint {
     /// A hash of the checkpoint contents.
     /// The EIP-191 compliant version of this hash is signed by validators.
     fn signing_hash(&self) -> H256 {
         // sign:
-        // domain_hash(mailbox_address, mailbox_domain) || root || index (as u32)
+        // domain_hash(mailbox_address, mailbox_domain) || root || index (as u32) || message_id
+        H256::from_slice(
+            Keccak256::new()
+                .chain(domain_hash(self.mailbox_address, self.mailbox_domain))
+                .chain(self.root)
+                .chain(self.index.to_be_bytes())
+                .finalize()
+                .as_slice(),
+        )
+    }
```

Comment:
Does `into` work instead of `as_slice` with the `from_slice`?

Reply:
nope